cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java

```diff
@@ -31,12 +31,41 @@ public interface CruiseControlApi {
      * @param host The address of the Cruise Control server.
      * @param port The port the Cruise Control Server is listening on.
      * @param options The rebalance parameters to be passed to the Cruise Control server.
-     * @param userTaskID This is the unique ID of a previous rebalance request. If a previous request had not been
+     * @param userTaskId This is the unique ID of a previous rebalance request. If a previous request had not been
      *                   completed when the response was returned then this ID can be used to retrieve the results of that
      *                   request.
      * @return A future for the rebalance response from the Cruise Control server containing details of the optimization.
      */
-    Future<CruiseControlRebalanceResponse> rebalance(String host, int port, RebalanceOptions options, String userTaskID);
+    Future<CruiseControlRebalanceResponse> rebalance(String host, int port, RebalanceOptions options, String userTaskId);
+
+    /**
+     * Send a request to the Cruise Control server to perform a cluster rebalance when adding new brokers.
+     * This method allows to move replicas from existing brokers to the new ones and avoids to run a full rebalance
+     * across all brokers in the cluster as done by {@link #rebalance(String, int, RebalanceOptions, String)}.
+     *
+     * @param host The address of the Cruise Control server.
+     * @param port The port the Cruise Control Server is listening on.
+     * @param options The add broker parameters to be passed to the Cruise Control server.
+     * @param userTaskId This is the unique ID of a previous addBroker request. If a previous request had not been
+     *                   completed when the response was returned then this ID can be used to retrieve the results of that
+     *                   request.
+     * @return A future for the rebalance response from the Cruise Control server containing details of the optimization.
+     */
+    Future<CruiseControlRebalanceResponse> addBroker(String host, int port, AddBrokerOptions options, String userTaskId);
```

Comment:
Should this not be `addBrokers` (plural) as you can add more than one.

Reply:
I had this doubt for days :-D ... but I decided to reflect the name of the `/add_broker` endpoint and keeping the API the same.